docs/using-hive.md

```diff
@@ -2,72 +2,231 @@
+
+```yaml
+apiVersion: v1
+data:
+  .dockerconfigjson: REDACTED
+kind: Secret
+metadata:
+  name: global-pull-secret
+  namespace: hive
+type: kubernetes.io/dockerconfigjson
+
+```
 
 ```bash
 oc edit hiveconfig hive
 ```
 
-The global pull secret name must be configured in the HiveConfig CR.
-
 ```yaml
 spec:
   globalPullSecret:
     name: global-pull-secret
 ```
 
-### Provisioning
+### OpenShift Version
 
-Hive supports two methods of specifying what version of OpenShift you wish to install. Most commonly you can create a ClusterImageSet which references an OpenShift 4 release image.
+Hive needs to know what version of OpenShift to install. You can specify an individual OpenShift release image for each cluster Hive creates, or you can use a global `ClusterImageSet` that will be used by all of the clusters Hive creates.
 
-An example ClusterImageSet:
+An example `ClusterImageSet`:
 
 ```yaml
 apiVersion: hive.openshift.io/v1
 kind: ClusterImageSet
 metadata:
-  name: openshift-v4.2.0
+  name: openshift-v4.3.0
 spec:
-  releaseImage: quay.io/openshift-release-dev/ocp-release:4.2.0
+  releaseImage: quay.io/openshift-release-dev/ocp-release:4.3.0
+```
+Alternatively you can specify an OpenShift 4 release image directly on your `ClusterDeployment`. This value will override any value set in your `ClusterImageSet`.
+
+### Cloud credentials
+
+Hive requires credentials to the cloud account into which it will install OpenShift clusters.
+
+#### AWS
+
+Create a `secret` containing your AWS access key and secret access key:
+
+```yaml
+apiVersion: v1
+data:
+  aws_access_key_id: REDACTED
+  aws_secret_access_key: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-aws-creds
+  namespace: hive
+type: Opaque
+```
+
+#### Azure
+
+Create a `secret` containing your Azure service principal:
+
+```yaml
+apiVersion: v1
+data:
+  osServicePrincipal.json: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-azure-creds
+  namespace: hive
+type: Opaque
+```
+
+#### GCP
+
+Create a `secret` containing your GCP service account key:
+
+```yaml
+apiVersion: v1
+data:
+  osServiceAccount.json: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-gcp-creds
+  namespace: hive
+type: Opaque
+```
+
+### SSH Key Pair
+
+(Optional) Hive uses the provided ssh key pair to ssh into the machines in the remote cluster. Hive connects via ssh to gather logs in the event of an installation failure. The ssh key pair is optional, but neither the user nor Hive will be able to ssh into the machines if it is not supplied.
+
+Create a Kubernetes secret containing a ssh key pair (typically generated with `ssh-keygen`)
+
+```yaml
+apiVersion: v1
+data:
+  ssh-privatekey: REDACTED
+  ssh-publickey: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-ssh-key
+  namespace: hive
+type: Opaque
+```
+
+### InstallConfig
+
+The OpenShift installer `InstallConfig` must be base64 encoded into a `secret` and referenced in the `ClusterDeployment`.
```

Comment:
"An OpenShift installer InstallConfig".

I would also note here that this is done to allow hive to support installing many versions of OpenShift.

Reply:
done